metricflow/model/parsing/dir_to_model.py

```diff
@@ -220,38 +219,47 @@ def parse_config_yaml(
+                elif document_type == DATA_SOURCE_TYPE:
+                    data_source_validator.validate(config_document[document_type])
+                    results.append(data_source_class.parse_obj(object_cfg))
+                elif document_type == MATERIALIZATION_TYPE:
+                    materialization_validator.validate(config_document[document_type])
+                    results.append(materialization_class.parse_obj(object_cfg))
+                else:
+                    issues.append(
+                        ValidationError(
+                            context=FileContext(file_name=ctx.filename, line_number=ctx.start_line),
+                            message=f"Invalid document type: {document_type}. Expected {DOCUMENT_TYPES}.",
+                        )
+                    )
+            # catches exceptions from jsonschema validator
+            except exceptions.ValidationError as e:
+                context = FileContext(file_name=ctx.filename, line_number=ctx.start_line)
                 issues.append(
                     ValidationError(
-                        context=FileContext(file_name=ctx.filename, line_number=ctx.start_line),
-                        message=f"Invalid document type: {document_type}. Expected {DOCUMENT_TYPES}.",
+                        context=context,
+                        message=f"YAML document did not conform to metric spec.\nError: {e}",
                     )
                 )
-    except exceptions.ValidationError as e:
-        context = FileContext(file_name=ctx.filename, line_number=ctx.start_line) if ctx is not None else None
-        issues.append(
-            ValidationError(
-                context=context,
-                message=f"YAML document did not conform to metric spec.\nError: {e}",
-            )
-        )
-    except ScannerError as e:
-        context = FileContext(file_name=ctx.filename, line_number=ctx.start_line) if ctx is not None else None
-        issues.append(ValidationError(context=context, message=str(e)))
-    except ParsingException as e:
-        context = FileContext(file_name=ctx.filename, line_number=ctx.start_line) if ctx is not None else None
-        issues.append(ValidationError(context=context, message=str(e)))
+            # catches exceptions from *.parse_obj calls
+            except ParsingException as e:
+                context = FileContext(file_name=ctx.filename, line_number=ctx.start_line)
+                issues.append(ValidationError(context=context, message=str(e)))
+            # general exception for a given document. Basicially we don't want an exception
+            # on one document to halt checking the rest of the documents
+            except Exception as e:
+                context = FileContext(file_name=ctx.filename, line_number=ctx.start_line)
+                issues.append(ValidationError(context=context, message=str(e)))
+    # If a runtime error occured, we still want this to break things
+    except RuntimeError:
```

Comment:
Interesting. Is this establishing our boundary between user input issues and internal errors?

I don't know if our error hierarchy is this disciplined, though. Even with Python - ValueError and AssertionErrors are not RuntimeError subclasses. I suggest we lose the global Exception -> issue conversion and instead refine around what user input might throw.

Reply:
Yes indeed. Basically internal errors aren't actually a validation error, or shouldn't be. If we swallow them, they might be harder to realize. Ideally if there is an internal logic error, it should be raised an exception and hopefully caught during unit tests or sanity testing changes.